### PR TITLE
Refactor the lifecycle test

### PR DESCRIPTION
**Pull request by yihuaf**
As previously discussed in #1827 , the `TestResult` is meant to be used in the test cases and not utility functions. Before this PR, the lifecycle test calls other tests using `TestResult` which is not inline with our intention. This PR fix this by refactoring all tests to not call other tests in test cases.

### CODE REVIEW

File: tests/rust-integration-tests/integration_test/src/tests/lifecycle/container_create.rs [hunk 1]
```diff
@@ -26,39 +26,45 @@ impl ContainerCreate {
...
-        match temp {
-            TestResult::Passed => TestResult::Failed(anyhow::anyhow!(
-                "Container should not have been created with same id, but was created."
-            )),
-            TestResult::Failed(_) => TestResult::Passed,
-            TestResult::Skipped => TestResult::Skipped,
+        match create::create(&self.project_path, &id) {
+            Ok(()) => {
+                let _ = kill::kill(&self.project_path, &id);
+                let _ = delete::delete(&self.project_path, &id);
```
Comment (YJDoc2) [inline tests/rust-integration-tests/integration_test/src/tests/lifecycle/container_create.rs:58]: This is duplicated in both arms, so we should extract it before the match.

File: tests/rust-integration-tests/integration_test/src/tests/lifecycle/container_create.rs [hunk 2]
```diff
@@ -26,39 +26,45 @@ impl ContainerCreate {
...
-            delete::delete(&self.project_path, &self.container_id);
+        match create::create(&self.project_path, &self.container_id) {
+            Ok(_) => {
+                let _ = kill::kill(&self.project_path, &self.container_id);
+                let _ = delete::delete(&self.project_path, &self.container_id);
+                TestResult::Passed
+            }
+            Err(_) => TestResult::Failed(anyhow::anyhow!(
+                "Container should have been created with valid id, but was not created."
+            )),
```
Comment (YJDoc2) [inline tests/rust-integration-tests/integration_test/src/tests/lifecycle/container_create.rs:47]: Maybe we should bubble the actual error up too (through the `anyhow!`), so if this ever fails, we can get the error

**Review by YJDoc2 — CHANGES_REQUESTED** [review]
Hey, overall its fine, just a few nits.

As you had said this makes much more sense that what we currently have. Thanks for taking initiative and making this change!

**Comment by yihuaf** [comment]
@YJDoc2 PTAL

**Review by YJDoc2 — APPROVED** [review]
lgtm :+1:

**Pull request merged**